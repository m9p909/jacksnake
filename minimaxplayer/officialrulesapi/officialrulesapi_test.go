package officialrulesapi_test

import (
	"jacksnake/minimaxplayer/officialrulesapi"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

/*
	type OfficialRulesAdapter interface {
		SimulateMove(board coreplayer.GameBoard, move string, snakeId string) coreplayer.GameBoard
		GetValidMoves(board coreplayer.GameBoard, snakeID string) []string
	}
*/
func Test_OfficialRulesAdapter(t *testing.T) {
	rules := officialrulesapi.GetOfficialRules()
	moves := rules.GetValidMoves(getCoreState(), "1")
	snaps.MatchSnapshot(t, moves)

}
