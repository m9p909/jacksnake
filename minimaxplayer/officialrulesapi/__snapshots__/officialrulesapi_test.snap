
[Test_OfficialRulesAdapter - 1]
[]string{"down", "left"}
---
